server/controller/users_controller.ts

```diff
@@ -122,3 +123,21 @@ export const handleCheckPassword = async (
     next(err);
   }
 };
+
+export const handleDeleteUser = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const userId = req.userId;
+    await deleteUser(userId);
+
+    res.clearCookie("accessToken");
+    res.clearCookie("refreshToken");
+    await deleteRefreshToken(userId);
+    res.status(200).json({ message: "회원탈퇴 성공" });
+  } catch (err: any) {
+    next(err);
+  }
+};
```

Comment:
@Siltarae
2개 이상의 DB 쿼리가 실행될때는 `트랜잭션`을 적용하여 모든 쿼리가 성공하거나 실패하도록 보장할 필요가 있습니다.
트랜잭션을 적용하면 User 삭제 쿼리까지 실행된 상태로 서버가 종료되어  RefreshToken이 삭제되지 못하는 경우를 방지할 수 있습니다. 

(지금은 둘 중 하나만 실패해도 치명적인 결함이 발생하지는 않겠지만요 😂)
```suggestion
  const connection = await pool.getConnection();
  try {
    await connection.beginTransaction();

    const userId = req.userId;
    await deleteUser(userId, connection);
    await deleteRefreshToken(userId, connection);

    res.clearCookie('accessToken');
    res.clearCookie('refreshToken');
    
    await connection.commit();
    res.status(200).json({ message: '회원탈퇴 성공' });
  } catch (err) {
    await connection.rollback();
    next(err);
  } finally {
    connection.release();
  }
};
```

Reply:
정처기 공부할때 봤던 트렌젝션을 이렇게 사용할 수 있겠네요. 좋은 리뷰 감사합니다